src/rgw/rgw_file.cc

```diff
@@ -1766,7 +1766,7 @@ void rgwfile_version(int *major, int *minor, int *extra)
   rc = new_fs->authorize(rgwlib.get_store());
   if (rc != 0) {
     delete new_fs;
-    return -EINVAL;
+    return rc;
```

Comment:
i cannot tell if this is the expected behavior as neither rgw_mount() nor rgw_mount2() comes with document explaining what their return value should be. the only thing i can refer is test. all of them expect "0" when the the call succeeds.